Cmek azure (#372)
neo4j/docs-aura: modules/ROOT/pages/platform/security/encryption.adoc

=== AWS automatic key rotation

Aura supports automatic key rotation via the AWS KMS.
To enable automatic key rotation in the AWS KMS, tick the *Key rotation* checkbox after initially creating a key, to automatically rotate the key once a year.

== Azure keys

=== Creating an Azure key vault

> CAUTION: It's crucial to differentiate between Azure Key Vault and Azure Key Vault Managed HSM for your key management needs. 
> This document focuses on using *Azure Key Vault*.
> 
> . In the Azure portal, create a link:https://learn.microsoft.com/en-us/azure/key-vault/general/[Key Vault]
> . In the *Basics tab* ensure the region matches your Aura database instance region and enable *purge protection*.
> . In the *Access configuration tab* enable *Azure role-based access control* and enable *Azure Disk Encryption for volume encryption*.
> . In the *Networking tab* allow access from all networks.
> . Select *Review + Create* and when the deployment is complete go to *resource* and record the *directory ID*, you need it later.

Reviewer: ```suggestion
[CAUTION]
====
It's crucial to differentiate between Azure Key Vault and Azure Key Vault Managed HSM for your key management needs.
This document focuses on using *Azure Key Vault*.
====

. Create a key vault in the Azure portal ensuring the region matches your Aura database instance region.
Enable *purge protection*, *Azure role-based access control*, and *Azure Disk Encryption for volume encryption*, and allow access from all networks.
. Once created and deployed, copy the *directory ID* from the *resource* tab, you need it later.
```

> ==== Add role assignment to the Key Vault
> 
> . Go to *Access control (IAM)* and add *role assignment*.
> . Search *Key Vault Administrator* and tap *Key Vault Administrator* - this will highlight the row in grey, and then press *next*.
> . Select members (search by name, and when you have added the relevant person, press *select*) and then *Review + Assign*.

Reviewer: ```suggestion
=== Add role assignment to the Key Vault

In the Azure portal, add role assignment by selecting relevant members as *Key Vault Administrator*.
```

=== Creating a key

Add a key in the Key Vault.

> When the key is created, record the *Key Identifier* because you need it later.

Reviewer: ```suggestion
When the key is created, record the *Key Identifier*, you need it later.
```

==== Create a Service Principal for the Neo4j CMK Application

In the Azure tenant where your key is located, create a Service Principal linked to the Neo4j CMK Application with the ID displayed in the Aura Console.

> For example, using link:https://learn.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-create-examples[Azure CLI] the command is:
> 
> [source,bash]
> ----
> `az ad sp create --id clientID`. 
> ----

Reviewer: ```suggestion
For example, using Azure CLI the command is:

[source,bash]
----
`az ad sp create --id clientID`.
----

See https://learn.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-create-examples[Azure documentation^] for more information.
```

==== Add role assignment to the key

> . Go to *Access control (IAM)* and *Add role assignment.*
> . Select *Key Vault Crypto Officer* to highlight the row in grey, and then press *next*.
> . Tap *Select member* and assign access to *User, group, or service principal*, and paste the Neo4j CMK Application ID and add it to *Selected members*, then *Review + Assign*.

Reviewer: ```suggestion
In the Azure portal, add role assignment by selecting relevant members as *Key Vault Crypto Officer*.
Use the Neo4j CMK Application ID from the Aura Console.
```
